### PR TITLE
Fix for `test_pty_no_color` when cache is cleared. NFC

### PR DESCRIPTION
**Pull request by sbc100**
Withouth this change the color message that emscripten generated when
it sets up the sysroot will cause the test to fail.

### CODE REVIEW

File: tests/test_other.py [hunk 1]
```diff
@@ -9858,14 +9858,18 @@ def test_build_error_color(self):
...
-    'fno_diagnostics_color': ['-fno-diagnostics-color'],
-    'fdiagnostics_color_never': ['-fdiagnostics-color=never'],
+    '': ['-fno-diagnostics-color'],
+    'never': ['-fdiagnostics-color=never'],
   })
   @no_windows('ptys and select are not available on windows')
   def test_pty_no_color(self, flag):
+    # Ensure that the sysroot is already constructed so that emcc doesn't
+    # output any log information in color.
+    self.emcc(test_file('hello_world.c'), ['-c'])
```
Comment (kripken) [inline tests/test_other.py:9868]: why is this needed? shouldn't the flag affect colors in the sysroot construction logging?

Comment (sbc100) [reply tests/test_other.py:9868]: (BTW we don't use python's logging mechaism for emcc error or warnings... it really only used for debugging and for niche cases like this where we report building of internal libraries).

So normal errors and warnings that emcc generates are not effected by this.

Comment (kripken) [reply tests/test_other.py:9868]: I see, makes sense. lgtm with an explanation of that in the test code.

**Comment by sbc100** [comment]
Looks like the fix for this depends on #17434
